llvm/lib/Analysis/ConstantFolding.cpp

```diff
@@ -2621,6 +2623,21 @@ static Constant *ConstantFoldLibCall2(StringRef Name, Type *Ty,
   switch (Func) {
   default:
     break;
+  case LibFunc_ldexp:
+  case LibFunc_ldexpf:
+  case LibFunc_ldexpl:
+  case LibFunc_scalbn:
+  case LibFunc_scalbnf:
+  case LibFunc_scalbnl:
+  case LibFunc_scalbln:
+  case LibFunc_scalblnf:
+  case LibFunc_scalblnl:
+    if (TLI->has(Func)) {
+      APFloat ret =
+          llvm::scalbn(Op1V, Op2V.convertToDouble(), RoundingMode::TowardZero);
```

Comment:
> There should be no conversion, this code is dead

I don't know how to remove the conversion here, it only accepts int type, causing the compilation to fail

Reply:
Sorry, please excuse my rudeness. Can you take any time to give me some guidance?

Reply:
This function is handling binary floating point calls, with 2 FP arguments. This operation does not follow this pattern. This code should be somewhere else. I don't see how this could would be reached as-is. The `dyn_cast<ConstantFP>` should always fail. This should go somewhere else for the FP + int, and match on a ConstantInt operand